cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs

```diff
@@ -171,3 +188,28 @@ runGovernanceDRepMetadataHashCmd
       . writeByteStringOutput mOutFile
       . serialiseToRawBytesHex
       $ metadataHash
+
+-- | Check the hash of the anchor data against the hash in the anchor if
+-- checkHash is set to CheckHash.
+carryHashChecks
+  :: (FetchURLError -> error)
+  -- ^ Function that takes a FetchURLError and returns the error type of the caller
+  -> (L.SafeHash L.StandardCrypto L.AnchorData -> L.SafeHash L.StandardCrypto L.AnchorData -> error)
+  -- ^ Function that takes the expected and actual hashes and returns the mismatch hash error type of the caller
+  -> PotentiallyCheckedAnchor DRepMetadataUrl (L.Anchor L.StandardCrypto)
+  -- ^ The information about anchor data and whether to check the hash (see 'PotentiallyCheckedAnchor')
+  -> ExceptT error IO ()
+carryHashChecks errorAdaptor hashMismatchError potentiallyCheckedAnchor =
+  let anchor = pcaAnchor potentiallyCheckedAnchor
```

Comment:
I'd put that in a `where` clause to reduce the level of indentation in the function.

Note that I usually refrain from subjective suggestions like this, but in this case, I find the body quite hard to read with the number of indentations (in a small number of lines).

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/commits/2e8b2b84fb9808012aa18ad1b702ea6223aa6c37